### ISSUE TITLE
questions about paired exon sequencing

### ISSUE THREAD

**Issue by jiaying2508** [CMUSchwartzLab/MosaicSim#1]
Hi, 
I have a few questions regarding the exonrunPairedSim function and wonder if you could provide some insights. 
1) The coverage was achieved when cov exceeds coverage using cov += 2*batch*subblock*ratio where ratio = 2*rl/fl. Can you briefly explain why it is necessary for the multiplication of 2 at both positions?
2) It seems to me that you are adding all the mutations for a tumor clone and creating a .gz file first then do the sampling on the exon regions using the provided file. How do you handle the fragments/reads for positions/regions where there is CNV or DEL before the position?
3) The coverage can be achieved by repeated sequencing for the exon regions. Does it mean each exon region will have similar coverage even if there is CNV or DEL events? i.e. will the generated reads differ in coverage such that copy number calling tools can be used to infer CNV using exon sequencing data?
Thank you in advance for helping with the questions!

**Comment by arjunsrivatsa** [comment]
1. I believe that the 2 in the ratio had to do with the fact that there are two paired reads corresponding to the read length that are covering the genome, the 2 in the coverage likely has to do with diploid genome corrections though I would need to more closely parse the code again for the rationale.
2. The exon sequencing has a locality assumption that the reads in the intervals are going to be closely positionally related to the original exon segments. If you don't believe this to be the case there is an option to use locality sensitive hashing in the code to find sequence similarity matches (though this portion has not been tested as thoroughly)
3. The base function will sequence in that interval on the mutated genome. If you use hashing only sequences that are matching sequence similarity to the original exon fragments would be returned. Whether CNVs or DELs are called likely depends widely on the tool you are using for calling and its assumptions. E.g. if you exploit paired normal assumptions on coverage the CNV caller might detect it or likewise the DEL caller.

**Comment by jiaying2508** [comment]
I noticed that the actual coverage for WES is less than the set coverage. The problem might be cause by calculating the ratio using 2*rl/fl instead of 2*rl/(interval[1]-inverval[0]).

btw, when sampling the tumor samples, is the tool just sampling the tumor clones? does the reference clone get sampled as well? Thanks!

**Comment by arjunsrivatsa** [comment]
Yes I think you are right, that value of ratio should be the fraction of the exon covered. 
The tool will sample all nodes including reference if you have "use_leaf_only = False" in the parameters file

**Comment by jiaying2508** [comment]
It seems like the reference node is not included. use_nodes is set to int_nodes when use_leaf_only os set to False. Do you think it make sense to set use_nodes = root_node instead?

**Comment by arjunsrivatsa** [comment]
The clones are being loaded from the "pickdclone" method. I thought that would include the root node, but it may not be, it is probably easily modified to include the root though.
